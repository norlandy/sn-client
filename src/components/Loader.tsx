import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles(() => ({
	root: {
		display: 'flex',
		justifyContent: 'center',
		margin: '20px 0',
		padding: '10px',
	},
}));

const CircularIndeterminate: React.FC = () => {
	const classes = useStyles();

	return (
		<aside className={classes.root}>
			<CircularProgress />
		</aside>
	);
};

export default CircularIndeterminate;
