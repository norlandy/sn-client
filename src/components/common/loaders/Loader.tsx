import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import CircularProgress, {CircularProgressProps} from '@material-ui/core/CircularProgress';

const useStyles = makeStyles(() => ({
	root: {
		display: 'flex',
		justifyContent: 'center',
	},
}));

type Props = CircularProgressProps & {disableMargin?: boolean};

const Loader: React.FC<Props> = ({disableMargin, ...props}) => {
	const classes = useStyles();

	return (
		<aside
			className={classes.root}
			style={{margin: disableMargin ? 0 : '20px 0', padding: disableMargin ? 0 : '10px'}}
		>
			<CircularProgress {...props} />
		</aside>
	);
};

export default Loader;
