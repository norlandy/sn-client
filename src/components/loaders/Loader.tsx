import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles(() => ({
	root: {
		display: 'flex',
		justifyContent: 'center',
	},
}));

type Props = {
	disableMargin?: boolean;
};

const Loader: React.FC<Props> = ({disableMargin}) => {
	const classes = useStyles();

	return (
		<aside
			className={classes.root}
			style={{margin: disableMargin ? 0 : '20px 0', padding: disableMargin ? 0 : '10px'}}
		>
			<CircularProgress />
		</aside>
	);
};

export default Loader;
