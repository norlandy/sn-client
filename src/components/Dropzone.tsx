import React, {useState} from 'react';
import {makeStyles} from '@material-ui/styles';
import BackupIcon from '@material-ui/icons/Backup';
import CardActionArea from '@material-ui/core/CardActionArea';
import DeleteIcon from '@material-ui/icons/Delete';
import Button from '@material-ui/core/Button';

import {ImageModal} from '@components/modals';
import Loader from '@components/Loader';

const useStyles = makeStyles({
	uploader: {
		display: 'flex',
		alignItems: 'center',
		marginBottom: 20,
	},
	imageWrapper: {
		maxWidth: 300,
		marginLeft: 20,
		cursor: 'pointer',
		lineHeight: 0,
		position: 'relative',
	},
	loadingBox: {
		position: 'absolute',
		bottom: 0,
		left: 0,
		width: '100%',
		height: '100%',
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
		borderBottomLeftRadius: 4,
		borderBottomRightRadius: 4,
		backgroundColor: 'rgba(0, 0, 0, .6)',
	},
	image: {
		width: '100%',
		height: '100%',
	},
});

type Props = {
	imagePreview: string;
	loadingImage: boolean;
	handleChangeImage: (e: React.ChangeEvent<HTMLInputElement>) => void;
	handleRemoveImage: () => void;
};

const Dropzone: React.FC<Props> = ({
	imagePreview,
	loadingImage,
	handleChangeImage,
	handleRemoveImage,
}) => {
	const classes = useStyles();

	const [imageModal, setImageModal] = useState(false);

	return (
		<div className={classes.uploader}>
			{imagePreview ? (
				<Button
					color='secondary'
					variant='contained'
					endIcon={<DeleteIcon />}
					onClick={handleRemoveImage}
					disabled={loadingImage}
				>
					Remove image
				</Button>
			) : (
				<Button
					color='primary'
					variant='contained'
					endIcon={<BackupIcon />}
					component='label'
					disabled={loadingImage}
				>
					Upload image
					<input type='file' style={{display: 'none'}} onChange={handleChangeImage} />
				</Button>
			)}

			{imagePreview && (
				<CardActionArea
					disabled={loadingImage}
					className={classes.imageWrapper}
					onClick={(): void => setImageModal(true)}
				>
					<img src={imagePreview} alt='' className={classes.image} />

					{loadingImage && (
						<div className={classes.loadingBox}>
							<Loader />
						</div>
					)}
				</CardActionArea>
			)}

			{imageModal && (
				<ImageModal image={imagePreview} closeModal={(): void => setImageModal(false)} />
			)}
		</div>
	);
};

export default Dropzone;
