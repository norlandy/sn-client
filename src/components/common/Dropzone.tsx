import React, {useState} from 'react';
import {makeStyles} from '@material-ui/styles';
import BackupIcon from '@material-ui/icons/Backup';
import CardActionArea from '@material-ui/core/CardActionArea';
import DeleteIcon from '@material-ui/icons/Delete';
import Button from '@material-ui/core/Button';

import {ImageModal} from '@components/common/modals';
import Loader from '@components/common/loaders/Loader';

const useStyles = makeStyles({
	uploader: {
		display: 'flex',
		alignItems: 'center',
		marginBottom: 20,
	},
	imageWrapper: {
		maxWidth: 400,
		height: 300,
		marginLeft: 20,
		cursor: 'pointer',
		lineHeight: 0,
		position: 'relative',
	},
	loadingBox: {
		position: 'absolute',
		bottom: 0,
		left: 0,
		width: '100%',
		height: '100%',
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
		borderBottomLeftRadius: 4,
		borderBottomRightRadius: 4,
		backgroundColor: 'rgba(0, 0, 0, .6)',
	},
	image: {
		width: '100%',
		height: '100%',
		objectFit: 'fill',
	},
});

type Props = {
	imagePreview: string;
	loadingImage: boolean;
	handleChangeImage: (e: React.ChangeEvent<HTMLInputElement>) => void;
	handleRemoveImage: () => void;
};

const Dropzone: React.FC<Props> = ({
	imagePreview,
	loadingImage,
	handleChangeImage,
	handleRemoveImage,
}) => {
	const classes = useStyles();

	const [imageModal, setImageModal] = useState(false);

	return (
		<div className={classes.uploader}>
			{imagePreview ? (
				<Button
					color='secondary'
					variant='contained'
					endIcon={<DeleteIcon />}
					onClick={handleRemoveImage}
					disabled={loadingImage}
				>
					Remove image
				</Button>
			) : (
				<Button
					color='primary'
					variant='contained'
					endIcon={<BackupIcon />}
					component='label'
					disabled={loadingImage}
				>
					Upload image
					<input type='file' style={{display: 'none'}} onChange={handleChangeImage} />
				</Button>
			)}

			{imagePreview && (
				<CardActionArea
					disabled={loadingImage}
					className={classes.imageWrapper}
					onClick={(): void => setImageModal(true)}
				>
					<img src={imagePreview} alt='' className={classes.image} />

					{loadingImage && (
						<div className={classes.loadingBox}>
							<Loader />
						</div>
					)}
				</CardActionArea>
			)}

			<ImageModal
				open={imageModal}
				image={imagePreview}
				closeModal={(): void => setImageModal(false)}
			/>
		</div>
	);
};

export default Dropzone;
