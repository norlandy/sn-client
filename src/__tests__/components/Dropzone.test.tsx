import React from 'react';
import {shallow} from 'enzyme';
import Button from '@material-ui/core/Button';
import CardActionArea from '@material-ui/core/CardActionArea';

import Dropzone from '@components/common/Dropzone';
import Loader from '@components/common/loaders/Loader';

describe(`${Dropzone.name} Component Test`, () => {
	it(`${Dropzone.name} match snapshot`, () => {
		const props = {
			imagePreview: '',
			loadingImage: false,
			handleChangeImage: () => {},
			handleRemoveImage: () => {},
		};
		const component = shallow(<Dropzone {...props} />);

		expect(component).toMatchSnapshot();
	});

	it(`${Dropzone.name} without imagePreview prop`, () => {
		const props = {
			imagePreview: '',
			loadingImage: false,
			handleChangeImage: jest.fn(),
			handleRemoveImage: () => {},
		};
		const component = shallow(<Dropzone {...props} />);

		expect(component.find(Button)).toHaveLength(1);
		expect(component.find(Button).text()).toEqual('Upload image');

		const changeEvent = {
			target: {
				value: 'some-image-url',
			},
		};
		component.find('input').simulate('change', changeEvent);
		expect(props.handleChangeImage).toHaveBeenCalledTimes(1);
		expect(props.handleChangeImage).toHaveBeenCalledWith(changeEvent);
	});

	it(`${Dropzone.name} with imagePreview prop`, () => {
		const props = {
			imagePreview: 'some-image',
			loadingImage: false,
			handleChangeImage: () => {},
			handleRemoveImage: jest.fn(),
		};
		const component = shallow(<Dropzone {...props} />);

		expect(component.find(Button)).toHaveLength(1);
		expect(component.find(Button).text()).toEqual('Remove image');

		expect(component.find(CardActionArea)).toHaveLength(1);
		expect(component.find(CardActionArea).prop('disabled')).toEqual(false);

		component.find(Button).simulate('click');
		expect(props.handleRemoveImage).toHaveBeenCalledTimes(1);
	});

	it(`${Dropzone.name} with loadingImage`, () => {
		const props = {
			imagePreview: 'some-image',
			loadingImage: true,
			handleChangeImage: () => {},
			handleRemoveImage: () => {},
		};
		const component = shallow(<Dropzone {...props} />);

		expect(component.find(Loader)).toHaveLength(1);
		expect(component.find(CardActionArea).prop('disabled')).toEqual(true);
	});
});
