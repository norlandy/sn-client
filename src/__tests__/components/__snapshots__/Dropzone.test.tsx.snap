// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropzone Component Test Dropzone match snapshot 1`] = `ShallowWrapper {}`;
